packages/ui/src/composables/useElementBackground/useElementBackground.ts

```diff
@@ -0,0 +1,67 @@
+import { ColorArray, parseRGBA } from './utils'
+import { ref, getCurrentInstance, Ref } from 'vue'
+import { useDomChangesObserver } from './useDomChangesObserver'
+
+type Maybe<T> = T | null | undefined
+
+const WHITE_COLOR_ARRAY: ColorArray = [255, 255, 255, 1]
+const ALPHA_ARRAY_INDEX = 3
+
+const rgba2hex = (rgba: number[]) => `#${rgba.map((n: number, i: number) => (i === 3 ? Math.round(n * 255) : n).toString(16).padStart(2, '0').replace('NaN', '')).join('')}`
+
+const isTransparent = (color: string) => color === 'rgba(0, 0, 0, 0)'
+
+const mixColors = (color1Array: ColorArray, color2Array: ColorArray, weight: number) => {
+  const newColor = []
+
+  for (let i = 0; i < 4; i++) {
+    newColor.push(Math.round((color1Array[i]) * (1 - weight) + (color2Array[i]) * weight))
```

Comment:
The `color1Array ` or `color2Array` arrays for a `rgb` color can be `[number, number, number]`. In this case `newColor` array will by `[number, number, number, undefined]`

Reply:
ColorArray can not be [number, number, number, undefined]

```ts
export type ColorArray = [number, number, number, number]
```